### PR TITLE
fix(pty): guard keep-alive ring-buffer replay against a closed socket

### PR DESCRIPTION
**Pull request by mindfury**
## Summary

The dashboard's keep-alive PTY terminal crashes its ASGI handler with an uncaught `WebSocketDisconnect` when the browser is closed while buffered output is being replayed on (re)attach. The operator sees a full uvicorn/starlette traceback printed to the terminal running `hermes dashboard`.

### Symptom (real report)

Closing the browser tab mid-session produced this on the `hermes dashboard` terminal:

```
ERROR:    Exception in ASGI application
...
  File ".../hermes_cli/web_server.py", line 15559, in pty_ws
    await session.attach(ws)
  File ".../hermes_cli/pty_session.py", line 93, in attach
    await ws.send_bytes(snap)
  File ".../starlette/websockets.py", line 169, in send_bytes
    await self.send({"type": "websocket.send", "bytes": data})
  ...
starlette.websockets.WebSocketDisconnect
```

## Root cause

`PtySession.attach()` replays the `RingBuffer` to the socket via `await ws.send_bytes(snap)` when a keep-alive terminal (`?attach=<token>`) attaches. That send had **no exception guard** — unlike the two sibling send sites in the same subsystem, both of which already wrap `send_bytes` in `try/except`:

- `PtySession._drain()` — `except Exception: pass` (keep buffering)
- `web_server._legacy_pump()` — `except Exception: return`

When the client socket is already gone by replay time (browser closed, half-open socket), starlette raises `WebSocketDisconnect`, which propagates uncaught out of `web_server.pty_ws` and up the ASGI stack.

Introduced in the keep-alive PTY feature (`e10e4bca8`). The test shipped with that feature never caught it: its `FakeBridge.read()` returns empty bytes, so the ring buffer is always empty and `send_bytes(snap)` is never called. Buffered output only exists in real use.

## Fix

Fixes the whole bug class at the source, plus a defensive guard at the call site:

- **`PtySession.attach()`** — wrap the replay send. On failure, roll the session back to a detached, reapable state (a dead socket must never look live and block the reaper) and re-raise so the caller can tear it down.
- **`web_server.pty_ws()`** — catch `WebSocketDisconnect` around `session.attach()` and detach the token cleanly instead of letting it escape the handler.

## Test plan

Adds `tests/test_pty_attach_disconnect_repro.py`:

- `test_pty_ws_attach_site_swallows_disconnect` — drives the **real** `pty_ws` handler; **fails on current `main`** with an uncaught `WebSocketDisconnect`, passes with the fix.
- `test_attach_raises_on_dead_socket_during_replay` — locks in the rollback-on-failure invariant (not attached, no live socket, buffer preserved).
- `test_attach_replays_buffer_on_live_socket` — happy-path replay still works.

Verified:

- [x] New repro test fails on clean `main`, passes with the fix (confirmed by stash/pop).
- [x] Existing PTY suite green: `tests/test_pty_session.py`, `tests/test_pty_keepalive_ws.py`, `tests/hermes_cli/test_web_server_pty_reconnect.py`, `tests/hermes_cli/test_web_server_pty_import.py` — 17 passed, 1 skipped (Windows-only).
- [x] The 3 pre-existing `TestPtyWebSocket` env failures (real PTY spawn under TestClient) fail identically with and without this change — untouched.

Note: the secondary Ctrl-C `KeyboardInterrupt` traceback in the original report is standard `asyncio.run` interrupt chaining and is intentionally left out of this focused fix.

### CODE REVIEW

**Comment by teknium1** [comment]
Thanks for the focused replay-disconnect regression coverage. The current `main` premise is valid: `PtySession.attach()` performs an unguarded buffered replay at `hermes_cli/pty_session.py:93`, and `pty_ws()` calls it before its existing `WebSocketDisconnect` handler at `hermes_cli/web_server.py:15788`.

### Problems
- The new rollback guarantee is limited to replay. `PtySession._drain()` still catches a failed live `ws.send_bytes(chunk)` at `hermes_cli/pty_session.py:74-79` without detaching the socket. Because `PtySessionRegistry.reap_idle()` requires `attached == false` (`hermes_cli/pty_session.py:173-182`), that failed socket can remain nonreapable until separate handler cleanup occurs.

### Suggested changes
- Detach the locally captured `ws` in `_drain()` on send failure, then add a drain-path regression test. The identity guard in `PtySession.detach()` (`hermes_cli/pty_session.py:95-105`) makes this safe if a newer socket has attached.

Automated hermes-sweeper review.
